File: python/my_first_package/my_first_package/services/my_first_service/structures.py
from ...schemas.my_first_module import structures as my_first_package_schemas_my_first_module_structures
from typelayer_common import documentation as typelayer_common_documentation
from dataclasses import dataclass
from typing import List, Dict, Optional


@dataclass
class MyFirstFunctionInput:
    my_first_argument: str

@dataclass
class MyFirstFunctionOutput:
    my_first_return: my_first_package_schemas_my_first_module_structures.MyFirstStructure

@dataclass
class MySecondFunctionInput:
    arg_one: my_first_package_schemas_my_first_module_structures.MyEnumStructure
    arg_two: str

@dataclass
class MySecondFunctionOutput:
    ret_one: List[str]

def default_my_first_function_input() -> MyFirstFunctionInput:

    my_first_argument_val = ""
    return MyFirstFunctionInput(my_first_argument=my_first_argument_val)

def describe_my_first_function_input() -> typelayer_common_documentation.StructureInfo:
    
    
    
    my_first_argument_field_info = typelayer_common_documentation.TlText()
    my_first_function_input_my_first_argument_field_info = typelayer_common_documentation.FieldDescription(name="my_first_argument", description=None, type_info=my_first_argument_field_info)
    my_first_function_input_constructor_info = typelayer_common_documentation.ConstructorInfo(name="my_first_function_input", description=None, fields=[my_first_function_input_my_first_argument_field_info])
    return typelayer_common_documentation.StructureInfo(structure_id="my_first_function_input", structure_type=typelayer_common_documentation.StructureEnum(), description=None, constructors=[my_first_function_input_constructor_info])

def default_my_first_function_output() -> MyFirstFunctionOutput:

    my_first_return_val = my_first_package_schemas_my_first_module_structures.default_my_first_structure()
    return MyFirstFunctionOutput(my_first_return=my_first_return_val)

def describe_my_first_function_output() -> typelayer_common_documentation.StructureInfo:
    
    
    
    my_first_return_field_info = typelayer_common_documentation.TlStructure(package="my_first_package", schema="structures", structure_id="my_first_structure")
    my_first_function_output_my_first_return_field_info = typelayer_common_documentation.FieldDescription(name="my_first_return", description=None, type_info=my_first_return_field_info)
    my_first_function_output_constructor_info = typelayer_common_documentation.ConstructorInfo(name="my_first_function_output", description=None, fields=[my_first_function_output_my_first_return_field_info])
    return typelayer_common_documentation.StructureInfo(structure_id="my_first_function_output", structure_type=typelayer_common_documentation.StructureEnum(), description=None, constructors=[my_first_function_output_constructor_info])

def default_my_second_function_input() -> MySecondFunctionInput:
    
    
    arg_one_val = my_first_package_schemas_my_first_module_structures.default_my_enum_structure()
    arg_two_val = ""
    return MySecondFunctionInput(arg_one=arg_one_val, arg_two=arg_two_val)

def describe_my_second_function_input() -> typelayer_common_documentation.StructureInfo:
    
    
    
    
    
    arg_one_field_info = typelayer_common_documentation.TlStructure(package="my_first_package", schema="structures", structure_id="my_enum_structure")
    my_second_function_input_arg_one_field_info = typelayer_common_documentation.FieldDescription(name="arg_one", description=None, type_info=arg_one_field_info)
    arg_two_field_info = typelayer_common_documentation.TlText()
    my_second_function_input_arg_two_field_info = typelayer_common_documentation.FieldDescription(name="arg_two", description=None, type_info=arg_two_field_info)
    my_second_function_input_constructor_info = typelayer_common_documentation.ConstructorInfo(name="my_second_function_input", description=None, fields=[my_second_function_input_arg_one_field_info, my_second_function_input_arg_two_field_info])
    return typelayer_common_documentation.StructureInfo(structure_id="my_second_function_input", structure_type=typelayer_common_documentation.StructureEnum(), description=None, constructors=[my_second_function_input_constructor_info])

def default_my_second_function_output() -> MySecondFunctionOutput:

    ret_one_val = []
    return MySecondFunctionOutput(ret_one=ret_one_val)

def describe_my_second_function_output() -> typelayer_common_documentation.StructureInfo:
    
    
    
    
    ret_one_inner_field_info = typelayer_common_documentation.TlText()
    ret_one_field_info = typelayer_common_documentation.TlList(list_inner=ret_one_inner_field_info)
    my_second_function_output_ret_one_field_info = typelayer_common_documentation.FieldDescription(name="ret_one", description=None, type_info=ret_one_field_info)
    my_second_function_output_constructor_info = typelayer_common_documentation.ConstructorInfo(name="my_second_function_output", description=None, fields=[my_second_function_output_ret_one_field_info])
    return typelayer_common_documentation.StructureInfo(structure_id="my_second_function_output", structure_type=typelayer_common_documentation.StructureEnum(), description=None, constructors=[my_second_function_output_constructor_info])

File: typescript/my-first-package/bin/services/myFirstService/structures.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
// Module: services.myFirstService.structures
const myFirstPackageSchemasMyFirstModuleStructures = __importStar(require("../../schemas/myFirstModule/structures"));
const typelayerCommonDocumentation = __importStar(require("typelayer-common"));
class MyFirstFunctionInput {
    constructor(arg) {
        Object.assign(this, arg);
    }
}
exports.MyFirstFunctionInput = MyFirstFunctionInput;
class MyFirstFunctionOutput {
    constructor(arg) {
        Object.assign(this, arg);
    }
}
exports.MyFirstFunctionOutput = MyFirstFunctionOutput;
class MySecondFunctionInput {
    constructor(arg) {
        Object.assign(this, arg);
    }
}
exports.MySecondFunctionInput = MySecondFunctionInput;
class MySecondFunctionOutput {
    constructor(arg) {
        Object.assign(this, arg);
    }
}
exports.MySecondFunctionOutput = MySecondFunctionOutput;
function defaultMyFirstFunctionInput() {
    let myFirstArgumentVal;
    myFirstArgumentVal = "";
    return new MyFirstFunctionInput({ myFirstArgument: myFirstArgumentVal });
}
exports.defaultMyFirstFunctionInput = defaultMyFirstFunctionInput;
function describeMyFirstFunctionInput() {
    let myFirstArgumentFieldInfo;
    let myFirstFunctionInputConstructorInfo;
    let myFirstFunctionInputMyFirstArgumentFieldInfo;
    myFirstArgumentFieldInfo = new typelayerCommonDocumentation.TlText({});
    myFirstFunctionInputMyFirstArgumentFieldInfo = new typelayerCommonDocumentation.FieldDescription({
        name: "my_first_argument",
        description: null,
        typeInfo: myFirstArgumentFieldInfo
    });
    myFirstFunctionInputConstructorInfo = new typelayerCommonDocumentation.ConstructorInfo({
        name: "my_first_function_input",
        description: null,
        fields: [myFirstFunctionInputMyFirstArgumentFieldInfo]
    });
    return new typelayerCommonDocumentation.StructureInfo({
        structureId: "my_first_function_input",
        structureType: new typelayerCommonDocumentation.StructureEnum({}),
        description: null,
        constructors: [myFirstFunctionInputConstructorInfo]
    });
}
exports.describeMyFirstFunctionInput = describeMyFirstFunctionInput;
function defaultMyFirstFunctionOutput() {
    let myFirstReturnVal;
    myFirstReturnVal = myFirstPackageSchemasMyFirstModuleStructures.defaultMyFirstStructure();
    return new MyFirstFunctionOutput({ myFirstReturn: myFirstReturnVal });
}
exports.defaultMyFirstFunctionOutput = defaultMyFirstFunctionOutput;
function describeMyFirstFunctionOutput() {
    let myFirstFunctionOutputConstructorInfo;
    let myFirstFunctionOutputMyFirstReturnFieldInfo;
    let myFirstReturnFieldInfo;
    myFirstReturnFieldInfo = new typelayerCommonDocumentation.TlStructure({
        package: "my_first_package",
        schema: "structures",
        structureId: "my_first_structure"
    });
    myFirstFunctionOutputMyFirstReturnFieldInfo = new typelayerCommonDocumentation.FieldDescription({
        name: "my_first_return",
        description: null,
        typeInfo: myFirstReturnFieldInfo
    });
    myFirstFunctionOutputConstructorInfo = new typelayerCommonDocumentation.ConstructorInfo({
        name: "my_first_function_output",
        description: null,
        fields: [myFirstFunctionOutputMyFirstReturnFieldInfo]
    });
    return new typelayerCommonDocumentation.StructureInfo({
        structureId: "my_first_function_output",
        structureType: new typelayerCommonDocumentation.StructureEnum({}),
        description: null,
        constructors: [myFirstFunctionOutputConstructorInfo]
    });
}
exports.describeMyFirstFunctionOutput = describeMyFirstFunctionOutput;
function defaultMySecondFunctionInput() {
    let argOneVal;
    let argTwoVal;
    argOneVal = myFirstPackageSchemasMyFirstModuleStructures.defaultMyEnumStructure();
    argTwoVal = "";
    return new MySecondFunctionInput({ argOne: argOneVal, argTwo: argTwoVal });
}
exports.defaultMySecondFunctionInput = defaultMySecondFunctionInput;
function describeMySecondFunctionInput() {
    let argOneFieldInfo;
    let argTwoFieldInfo;
    let mySecondFunctionInputArgOneFieldInfo;
    let mySecondFunctionInputArgTwoFieldInfo;
    let mySecondFunctionInputConstructorInfo;
    argOneFieldInfo = new typelayerCommonDocumentation.TlStructure({
        package: "my_first_package",
        schema: "structures",
        structureId: "my_enum_structure"
    });
    mySecondFunctionInputArgOneFieldInfo = new typelayerCommonDocumentation.FieldDescription({ name: "arg_one", description: null, typeInfo: argOneFieldInfo });
    argTwoFieldInfo = new typelayerCommonDocumentation.TlText({});
    mySecondFunctionInputArgTwoFieldInfo = new typelayerCommonDocumentation.FieldDescription({ name: "arg_two", description: null, typeInfo: argTwoFieldInfo });
    mySecondFunctionInputConstructorInfo = new typelayerCommonDocumentation.ConstructorInfo({
        name: "my_second_function_input",
        description: null,
        fields: [
            mySecondFunctionInputArgOneFieldInfo,
            mySecondFunctionInputArgTwoFieldInfo
        ]
    });
    return new typelayerCommonDocumentation.StructureInfo({
        structureId: "my_second_function_input",
        structureType: new typelayerCommonDocumentation.StructureEnum({}),
        description: null,
        constructors: [mySecondFunctionInputConstructorInfo]
    });
}
exports.describeMySecondFunctionInput = describeMySecondFunctionInput;
function defaultMySecondFunctionOutput() {
    let retOneVal;
    retOneVal = [];
    return new MySecondFunctionOutput({ retOne: retOneVal });
}
exports.defaultMySecondFunctionOutput = defaultMySecondFunctionOutput;
function describeMySecondFunctionOutput() {
    let mySecondFunctionOutputConstructorInfo;
    let mySecondFunctionOutputRetOneFieldInfo;
    let retOneFieldInfo;
    let retOneInnerFieldInfo;
    retOneInnerFieldInfo = new typelayerCommonDocumentation.TlText({});
    retOneFieldInfo = new typelayerCommonDocumentation.TlList({
        listInner: retOneInnerFieldInfo
    });
    mySecondFunctionOutputRetOneFieldInfo = new typelayerCommonDocumentation.FieldDescription({ name: "ret_one", description: null, typeInfo: retOneFieldInfo });
    mySecondFunctionOutputConstructorInfo = new typelayerCommonDocumentation.ConstructorInfo({
        name: "my_second_function_output",
        description: null,
        fields: [mySecondFunctionOutputRetOneFieldInfo]
    });
    return new typelayerCommonDocumentation.StructureInfo({
        structureId: "my_second_function_output",
        structureType: new typelayerCommonDocumentation.StructureEnum({}),
        description: null,
        constructors: [mySecondFunctionOutputConstructorInfo]
    });
}
exports.describeMySecondFunctionOutput = describeMySecondFunctionOutput;
